### PR TITLE
latte: fixes

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/bml/package.py [hunk 1]
```diff
@@ -38,3 +38,6 @@ class Bml(CMakePackage):
 
     depends_on("blas")
     depends_on("lapack")
+
+    def cmake_args(self):
+        return ['-DBUILD_SHARED_LIBS=ON']
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/bml/package.py]: Would it make sense to control this with a variant like in https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/ldc/package.py ? I think this will exclusively build the shared libs, vs. both shared and static.

File: var/spack/repos/builtin/packages/qmd-progress/package.py [hunk 2]
```diff
@@ -47,6 +47,7 @@ class QmdProgress(CMakePackage):
     def cmake_args(self):
         spec = self.spec
         args = ['-DCMAKE_Fortran_FLAGS=-ffree-line-length-none']
+        args.append('-DBUILD_SHARED_LIBS=ON')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/qmd-progress/package.py]: Perhaps this should also be optional?

File: var/spack/repos/builtin/packages/bml/package.py (hunk 1 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/bml/package.py]: cmake can only build static or shared, not both at the same time.

In an earlier review, I was told shared is prefered, I couldn't find the discussion anymore, but maybe @adamjstewart or @alalazo remember.

Comment (scheibelp) [reply var/spack/repos/builtin/packages/bml/package.py]: The original author @rspavel may have had use for the static libs (which I think is the default when `BUILD_SHARED_LIBS` is not specified). I'll say it would be easy to think about if this was added as an option.

> In an earlier review, I was told shared is prefered

Was this one of your PRs? In any case I think that may argue for making the default `shared` but it doesn't sound wrong to optionally build static.

**Comment by junghans** [comment]
@scheibelp: I made it into a variant.

File: var/spack/repos/builtin/packages/latte/package.py [hunk 3]
```diff
@@ -29,18 +29,30 @@
...
     depends_on("cmake@3.1:", type='build')
     depends_on('blas')
     depends_on('lapack')
+    depends_on('mpi', when='+mpi')
+    depends_on('qmd-progress', when='+progress')
 
     root_cmakelists_dir = 'cmake'
 
     def cmake_args(self):
         options = ['-DBUILD_SHARED_LIBS=ON']
```
Comment (becker33) [inline var/spack/repos/builtin/packages/latte/package.py]: While you're editing, it would be great if you added a `shared` variant here. It can also be a separate PR.

File: var/spack/repos/builtin/packages/qmd-progress/package.py [hunk 4]
```diff
@@ -47,6 +48,8 @@ class QmdProgress(CMakePackage):
     def cmake_args(self):
         spec = self.spec
         args = ['-DCMAKE_Fortran_FLAGS=-ffree-line-length-none']
+        if '+shared' in spec:
+            args.append('-DBUILD_SHARED_LIBS=ON')
```
Comment (becker33) [inline var/spack/repos/builtin/packages/qmd-progress/package.py]: Please make this add `-DBUILD_SHARED_LIBS=OFF` in the case where `~shared` in spec, as you did for bml.

Comment (junghans) [reply var/spack/repos/builtin/packages/qmd-progress/package.py]: `-DBUILD_SHARED_LIBS=OFF` is the default, but sure.

File: var/spack/repos/builtin/packages/latte/package.py [hunk 5]
```diff
@@ -29,18 +29,34 @@
...
     def cmake_args(self):
-        options = ['-DBUILD_SHARED_LIBS=ON']
+        options = []
+        if '+shared' in spec:
+            options.append('-DBUILD_SHARED_LIBS=ON')
+        else:
+            options.append('-DBUILD_SHARED_LIBS=OFF')
+        if '+mpi' in self.spec:
+            options.append('-DO_MPI=yes')
+        if '+progres' in self.spec:
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/latte/package.py]: Should this be `if '+progress' in self.spec`?

Comment (junghans) [reply var/spack/repos/builtin/packages/latte/package.py]: Good catch.

**Comment by scheibelp** [comment]
LGTM - I'll get a chance to merge this in a few hours

**Comment by scheibelp** [comment]
Thanks!
